### PR TITLE
wttrbar: init at 0.3.2

### PR DESCRIPTION
**Pull request by khaneliman**
###### Description of changes
Adding new package to repo. https://github.com/bjesus/wttrbar is a Waybar module for displaying weather information.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/wttrbar/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "bjesus";
+    repo = pname;
+    rev = version;
+    hash = "sha256-RQeRDu8x6OQAD7VYT7FwBfj8gxn1nj6hP60oCIiuAgg=";
+  };
+
+  cargoHash = "sha256-hJCEA6m/iZuSjWRbbaoJ5ryG0z5U/IWhbEvNAohFyjg=";
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/wttrbar/default.nix:19]:
```suggestion
  meta = {
```

File: pkgs/applications/misc/wttrbar/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    description = "A simple but detailed weather indicator for Waybar using wttr.in.";
+    homepage = "https://github.com/bjesus/wttrbar";
+    license = licenses.mit;
```
Comment (AndersonTorres) [inline pkgs/applications/misc/wttrbar/default.nix:22]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/applications/misc/wttrbar/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    maintainers = with maintainers; [khaneliman];
```
Comment (AndersonTorres) [inline pkgs/applications/misc/wttrbar/default.nix:23]:
```suggestion
    maintainers = with lib.maintainers; [ khaneliman ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -41590,4 +41590,6 @@ with pkgs;
   waylyrics = callPackage ../applications/audio/waylyrics { };
 
   gitrs = callPackage ../tools/misc/gitrs { };
+
+  wttrbar = callPackage ../applications/misc/wttrbar { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:41594]:
```suggestion
  wttrbar = callPackage ../applications/misc/wttrbar { };

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/wttrbar/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/wttrbar/default.nix:5]:
```suggestion

rustPlatform.buildRustPackage rec {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
Ping me tomorrow to merge

File: pkgs/applications/misc/wttrbar/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "bjesus";
+    repo = "wttrbar";
+    rev = version;
+    hash = "sha256-RQeRDu8x6OQAD7VYT7FwBfj8gxn1nj6hP60oCIiuAgg=";
+  };
+
+  cargoHash = "sha256-hJCEA6m/iZuSjWRbbaoJ5ryG0z5U/IWhbEvNAohFyjg=";
+
+  meta = {
+    description = "A simple but detailed weather indicator for Waybar using wttr.in.";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/wttrbar/default.nix:20]:
```suggestion
    description = "A simple but detailed weather indicator for Waybar using wttr.in";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by khaneliman** [comment]
@AndersonTorres Should be good, now.

**Review by AndersonTorres — APPROVED** [review]

**Comment by khaneliman** [comment]
Fixed to build on darwin

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by khaneliman** [comment]
@AndersonTorres  fourth times the charm? :P

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**